server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -609,16 +609,22 @@ public String toString() {
 
         private abstract class TimeUnitPreparedRounding extends PreparedRounding {
             @Override
-            public double roundingSize(long utcMillis, DateTimeUnit timeUnit) {
+            public double roundingSize(Long utcMillis, DateTimeUnit timeUnit) {
                 if (timeUnit.isMillisBased == unit.isMillisBased) {
```

Comment:
Any chance you could group this so `unit.isMillisBased` is called once? I think it might be a bit easier to read if you juggle the "arms" of the `if` statement.